gno.land/pkg/sdk/vm/params.go

```diff
@@ -21,19 +24,16 @@ var ASCIIDomain = regexp.MustCompile(`^(?:[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-
 type Params struct {
 	SysNamesPkgPath string `json:"sysnames_pkgpath" yaml:"sysnames_pkgpath"`
 	ChainDomain     string `json:"chain_domain" yaml:"chain_domain"`
-}
-
-// NewParams creates a new Params object
-func NewParams(namesPkgPath, chainDomain string) Params {
-	return Params{
-		SysNamesPkgPath: namesPkgPath,
-		ChainDomain:     chainDomain,
-	}
+	ValsetRealmPath string `json:"valset_realm_path" yaml:"valset_realm_path"`
 }
 
 // DefaultParams returns a default set of parameters.
 func DefaultParams() Params {
-	return NewParams(sysNamesPkgDefault, chainDomainDefault)
+	return Params{
+		SysNamesPkgPath: sysNamesPkgDefault,
+		ChainDomain:     chainDomainDefault,
+		ValsetRealmPath: ValsetRealmDefault,
```

Comment:
note -> probably we don't need this, but let's wait for the end of this PR.

Reply:
We need this because params are stored as part of the realm, not as sys params (which can _only_ be modified with a GovDAO proposal, not through the valopers realm) -- so we need to know which realm they are stored with.

Sys params short circuit if they're not updated directly from the `gno.land/r/sys/params` realm:
```go
func assertSysParamsRealm(m *gno.Machine) {
	// XXX improve
	if len(m.Frames) < 2 {
		panic("should not happen")
	}
	if m.Frames[len(m.Frames)-1].LastPackage.PkgPath != "sys/params" {
		panic("should not happen")
	}
	if m.Frames[len(m.Frames)-2].LastPackage.PkgPath != "gno.land/r/sys/params" {
		// XXX this should not happen after import rule.
		panic(`"sys/params" can only be used from "gno.land/r/sys/params"`)
	}
}
```